src/engine/sparqlExpressions/UuidExpressions.h

```diff
@@ -0,0 +1,73 @@
+//  Copyright 2024, University of Freiburg,
+//                  Chair of Algorithms and Data Structures
+//  Author: Hannes Baumann <baumannh@informatik.uni-freiburg.de>
+
+// Test for UuidExpression can be found in RandomExpressionTest.cpp
+
+#pragma once
+
+#include "engine/sparqlExpressions/SparqlExpression.h"
+#include "util/ChunkedForLoop.h"
+#include "util/Random.h"
+
+namespace sparqlExpression {
+
+using LiteralOrIri = ad_utility::triple_component::LiteralOrIri;
+
+inline constexpr auto fromLiteral = [](std::string_view str) {
+  return LiteralOrIri{
+      ad_utility::triple_component::Literal::literalWithNormalizedContent(
+          asNormalizedStringViewUnsafe(str))};
+};
+
+inline constexpr auto fromIri = [](std::string_view str) {
+  return LiteralOrIri{
+      ad_utility::triple_component::Iri::fromStringRepresentation(
+          absl::StrCat("<urn:uuid:"sv, str, ">"sv))};
+};
+
+inline constexpr auto litUuidKey = [](int64_t randId) {
+  return absl::StrCat("STRUUID "sv, std::to_string(randId));
+};
+
+inline constexpr auto iriUuidKey = [](int64_t randId) {
+  return absl::StrCat("UUID "sv, std::to_string(randId));
+};
+
+// With UuidExpression<fromIri, iriUuidKey>, the UUIDs are returned as an
+// Iri object: <urn:uuid:b9302fb5-642e-4d3b-af19-29a8f6d894c9> (example). With
+// UuidExpression<fromLiteral,, litUuidKey>, the UUIDs are returned as as an
+// Literal object: "73cd4307-8a99-4691-a608-b5bda64fb6c1" (example).
+template <auto FuncConv, auto FuncKey>
+class UuidExpression : public SparqlExpression {
```

Comment:
Best also move this class (which has certain preconditions on its template parameters) into the mentioned internal namespace and call it  `UuidExpressionImpl`. 

For the public `sparqlExpression` namespace just add a `using UuidExpression = UuiddExpressionImpl<detailbla::...>`, same for the `StrUuidExpression`. This makes the interface much cleaner andd more encapsulated.